libdevcore/CommonTime.h

```diff
@@ -0,0 +1,34 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file CommonTime.h
+ * @author Lefteris Karapetsas <lefteris@ethdev.com>
+ * @date 2015
+ *
+ * Helper functions having to do with time
+ */
+
+#pragma once
+
+#include <time.h>
+
+namespace dev
+{
+
+/// Cross platform wrapper for thread safe gmtime()
```

Comment:
It's thread safe in the way that we provide the result and the static storage of the implementation of the function is not used. It's the difference between `gmtime()` and `gmtime_r()`.

For more info regarding this you can check [this](http://stackoverflow.com/a/25618891/110395) SO post